### PR TITLE
[PAS289][Defect] Feedback no longer a confirmation step when deleting user credentials

### PR DESCRIPTION
**Pull request by jonashendrickx**
Using the confirmation dialog created by Anders to show the modal. To be able to use the reusable Razor component `ConfirmEditForm`, it was required to migrate the partial view to a Razor component as well.

The form submission is intercepted. When confirming, the form submission is allowed to execute. Upon successfully deleting the credential, the page is refreshed.

Added additional validation to ensure the credentialId is a 'base64url' encoded string.

**The alternative solution** would be to apply the attributes to the form and the button directly. But you would lose type safety. If the behavior happens to be altered, we'd have to hunt down every single usage of the confirmation modal.

### CODE REVIEW

File: src/AdminConsole/Pages/App/Credentials/User.cshtml.cs [hunk 1]
```diff
@@ -22,24 +21,16 @@ public class UserModel : PageModel
...
+        Credentials = await _passwordlessClient.ListCredentialsAsync(UserId);
         Aliases = await _passwordlessClient.ListAliasesAsync(UserId);
     }
 
-    public async Task<IActionResult> OnPost(string token)
-    {
-        var res = await _passwordlessClient.VerifyTokenAsync(token);
-        return new JsonResult(res);
-    }
-
```
Comment (abergs) [inline src/AdminConsole/Pages/App/Credentials/User.cshtml.cs:42]: question: Why was this removed? Is it no longer needed?

Comment (jonashendrickx) [reply src/AdminConsole/Pages/App/Credentials/User.cshtml.cs:42]: It was not used at all. Might have been a left-over from when the playground was developed. I'll double check again.

File: src/AdminConsole/Components/Shared/Credentials.razor.cs [hunk 2]
```diff
@@ -1,52 +1,83 @@
...
+using Microsoft.AspNetCore.Components;
+using Passwordless.AdminConsole.Helpers;
 using Passwordless.Common.Extensions;
+using Passwordless.Common.Validation;
 
-namespace Passwordless.AdminConsole.Pages.Shared;
+namespace Passwordless.AdminConsole.Components.Shared;
 
-public sealed class CredentialsModel
+public partial class Credentials : ComponentBase
```
[the remark this answers is not in the document]
Comment (jonashendrickx) [reply src/AdminConsole/Components/Shared/Credentials.razor.cs:9]: If you split the `@code` section into a separate file from `Credentials.razor` it would be named `Credentials.razor.cs`. But you have to make the class partial: https://learn.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-8.0#partial-class-support

File: src/AdminConsole/Pages/App/Credentials/User.cshtml.cs (hunk 1 above)
Comment (jonashendrickx) [reply src/AdminConsole/Pages/App/Credentials/User.cshtml.cs:42]: @abergs No other references are found. Seems ok to go

File: src/AdminConsole/Components/Shared/Credentials.razor.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (abergs) [reply src/AdminConsole/Components/Shared/Credentials.razor.cs:9]: OK

**Review by abergs — APPROVED** [review]
